lib/rules/no-semantic-errors.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Enforces that there is no semantic and syntactic errors
+ * @author Armano <https://github.com/armano2>
+ */
+"use strict";
+
+const util = require("../util");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "Enforces that there is no semantic and syntactic errors",
+            category: "TypeScript",
+            url: util.metaDocsUrl("no-semantic-errors"),
+        },
+        schema: [],
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+
+        const program =
+            context.parserServices && context.parserServices.program;
+
+        if (!program) {
+            return [];
```

Comment:
should we print error if there is no `parserServices`?

Reply:
I don’t think so. If the user has a mixed TS and JS codebase, they might have some files without `parserServices`. I might be wrong, though.

Reply:
parserServices is available for js files if you are parsing it with eslint-plugin-typescript

Reply:
Depends on what the likelihood of there not being parserServices?
Isn't that required by the parser, and thus a mandatory thing?

Reply:
There is few cases when parserServices is not present:
- there are missing fields in configuration
   ```js
   { 
         generateServices: true,
         project: "./tsconfig.json",
         tsconfigRootDir: __dirname // this field is optional
   }
   ```
- you are using wrong parser (eg. `babel-eslint`)

Reply:
Well considering that we're going to build more rules based on type information, I would make a utility function to fetch the services from the context, which throws if there is no services.

```TS
const program = getParserServices(context);
//////
function getParserServices(context) {
  if (context.parserServices == null) {
    throw new Error('the world is on fire');
  }
  return context.parserServices;
}
```

Reply:
i like this idea, but i will do few more check there